### DISCUSSION TITLE
Laravel Authorization disappear when component is refreshed

### DISCUSSION THREAD

**Discussion by krisnaw** [livewire/livewire#3545] [1. Help]
So I have 2 livewire components. Let's called it Component A and Component B. 
On component B  I have an action that then triggers an Event. 

**Blade view of Component B**
```
<button wire:click="save">Save</button>
```

```
public function save()
{
  $this->emit('postAdded');
}
```

On component A I have a listener that will render the component like this   `$this->render();`

```
protected $listeners = ['postAdded' => '$refresh'];

```

**Blade view of Component A**
```
<p>  Count number {{ $postCount }} </p>
@can('subscribed')
  <button>Delete count</button>
@endcan
```

**Issues:** 
When component A is refreshed, the Blade directive @can for the Authorization doesn't get refreshed. 

So it became like this
 **Blade view of Component A**
```
<p>  Count number {{ $postCount }} </p>
```

The button inside @can blade directive is disappeared.

**Comment by joshhanley** [comment]
@krisnaw Livewire requires there to be only a single root element, see here in the docs https://laravel-livewire.com/docs/2.x/rendering-components#render-method
> Make sure your Blade view only has ONE root element.

Try wrapping your "Component A" blade view in a div, and it should then work.

Hope this helps!

**Reply by krisnaw** [reply]
Hi @joshhanley thanks for the reply. 

Yes, it was wrapped on a single component. Otherwise, it won't render. The issue happens when the "Component A" is refresh after listen event from "Component B"

Blade view of "Component A" before the refresh. 
```
<div>
    <p>  Count number {{ $postCount }} </p>
    @can('subscribed')
      <button>Delete count</button>
    @endcan
</div>
```

Blade view of "Component A" after the refresh. 
```
<div>
    <p>  Count number {{ $postCount }} </p>
</div>
```

**Reply by joshhanley** [reply]
@krisnaw no worries! Wasn't in the example code and a lot of people run into issues with it not running because of that 🙂

Can you put some logging in your "subscribed" gate to see why it might be failing the second time?

**Reply by krisnaw** [reply]
Hi @joshhanley thanks again. 
Turns out the issue was the Gate itself. Since I put simple URL validation in it, it sends a different URL when the component is refreshed. Which caused the Gate to return false.

**Reply by joshhanley** [reply]
@krisnaw ah right yep! Yeah Livewire uses a custom route for communication 🙂
